bundles/pubsub/pubsub_protocol_wire_v1/src/pubsub_wire_protocol_impl.c

```diff
@@ -0,0 +1,309 @@
+    if (sync != PROTOCOL_WIRE_SYNC) {
+        status = CELIX_ILLEGAL_ARGUMENT;
+    } else {
+        unsigned int envelopeVersion;
+        idx = readInt(data, idx, &envelopeVersion);
+        if (envelopeVersion != PROTOCOL_WIRE_ENVELOPE_VERSION) {
+            status = CELIX_ILLEGAL_ARGUMENT;
+        } else {
+            idx = readInt(data, idx, &message->header.msgId);
+            idx = readShort(data, idx, &message->header.msgMajorVersion);
+            idx = readShort(data, idx, &message->header.msgMinorVersion);
+            idx = readInt(data, idx, &message->header.payloadSize);
+            readInt(data, idx, &message->header.metadataSize);
+        }
+    }
+
+    return status;
+}
+
+celix_status_t pubsubProtocol_decodePayload(void* handle, void *data, size_t length, pubsub_protocol_message_t *message){
+    message->payload.payload = data;
+    message->payload.length = length;
+
+    return CELIX_SUCCESS;
+}
+
+celix_status_t pubsubProtocol_decodeMetadata(void* handle, void *data, size_t length, pubsub_protocol_message_t *message) {
+    celix_status_t status = CELIX_SUCCESS;
+
+    uint32_t nOfElements;
+    size_t idx = readInt(data, 0, &nOfElements);
+    char *netstring = data + idx;
+    int netstringLen = length - idx;
+
+    message->metadata.metadata = celix_properties_create();
+    while (idx < length) {
+        size_t outlen;
+        status = pubsubProtocol_parseNetstring(netstring, netstringLen, &netstring, &outlen);
+        if (status != CELIX_SUCCESS) {
+            break;
+        }
+        char *key = strndup(netstring, outlen);
+        netstring += outlen + 1;
+        idx += outlen + 3;
+
+        status = pubsubProtocol_parseNetstring(netstring, netstringLen, &netstring, &outlen);
+        if (status != CELIX_SUCCESS) {
+            break;
+        }
+        char *value = strndup(netstring, outlen);
+        netstring += outlen + 1;
+        idx += outlen + 3;
+
+        celix_properties_setWithoutCopy(message->metadata.metadata, key, value);
+    }
+
+    return status;
+}
+
+static celix_status_t pubsubProtocol_createNetstring(const char* string, char** netstringOut) {
+    celix_status_t status = CELIX_SUCCESS;
+
+    size_t str_len = strlen(string);
+    if (str_len == 0) {
+        // 0:,
+        *netstringOut = calloc(1, 4);
+        if (*netstringOut == NULL) {
+            status = CELIX_ENOMEM;
+        } else {
+            *netstringOut[0] = '0';
+            *netstringOut[1] = ':';
+            *netstringOut[2] = ',';
+            *netstringOut[3] = '\0';
+        }
+    } else {
+        size_t numlen = ceil(log10(str_len + 1));
+        *netstringOut = calloc(1, numlen + str_len + 3);
+        if (*netstringOut == NULL) {
+            status = CELIX_ENOMEM;
+        } else {
+            sprintf(*netstringOut, "%zu:%s,", str_len, string);
+        }
+    }
+
+    return status;
+}
+
+/* Reads a netstring from a `buffer` of length `buffer_length`. Writes
```

Comment:
Update LICENSE file?

Reply:
Code comes from https://github.com/PeterScott/netstring-c, which does not list a license, and only mentions "public domain". S0 not needed I think?